pkg/services/object/delete/delete.go

```diff
@@ -3,11 +3,24 @@ package deletesvc
 import (
 	"context"
 
+	"github.com/nspcc-dev/neofs-node/pkg/services/object/util"
 	"go.uber.org/zap"
 )
 
 // Delete serves requests to remote the objects.
 func (s *Service) Delete(ctx context.Context, prm Prm) error {
+	// If session token is not found we will fail during tombstone PUT.
+	// Here we fail immediately to ensure no unnecessary network communication is done.
+	if tok := prm.common.SessionToken(); tok != nil {
```

Comment:
A few thoughts about it:

1. does that make object removing operation require a session token?
2. if `1.` is "yes", we need either remove such [branches](https://github.com/nspcc-dev/neofs-node/blob/5284ac53f906a0c1abd8f487245e077fa688c07a/pkg/services/object/delete/exec.go#L251-L255). if "no" (a node could remove public RW objects), then that change is incorrect

Reply:
incorrectly understood that change.